### ISSUE TITLE
Night Mode causes persistent screen problems (persist after the app is closed!)

### ISSUE THREAD

**Issue by ei14** [koreader/koreader#11928]
* KOReader version: 2024.04
* Device: Kobo Elipsa 2e 

#### Issue

Night mode does not work. When entered, the entire screen goes white. When I blindly tap the screen buttons, faint outlines of menu items appear, all overlapping with each other and several book pages.

When I stumble my way into disabling night mode, everything seems fine... But when I go to the native Kobo reader and go to dark mode there, the same white screen issue occurs.

The white issue does NOT occur in the native Kobo reader UNLESS I specifically (1) open koreader, (2) turn on night mode, (3) manage to turn it off, then (4) exit koreader. The issue persists until the device is power-cycled (sleep mode is not enough).

#### Steps to reproduce

(1) Enable night mode. Observe it does not work.

(2) Enable night mode, disable it, then return to the Kobo native reader. Observe dark mode does not work.

**Comment by tymmej** [comment]
You can try try nightly version, it seems there was some fixes: https://github.com/koreader/koreader-base/pull/1768

**Comment by NiLuJe** [comment]
Yup, current nightlies should be unaffected. (At the cost of not using the dedicated eclipse waveform modes, because they appear to be hella buggy. Newer MTK devices are mysteriously unaffected).

> But when I go to the native Kobo reader and go to dark mode there, the same white screen issue occurs.

That's an... interesting observation that might be useful for someone with an affected device to look into.

My running theory was that something, probably MDP related, was unhappy about handling the eclipse waveform modes @ 8bpp (a theory bolstered by the fact that the newer, unaffected devices, prohibit switching to 8bpp in the first place); but since we switch back to 32bpp for Nickel on exit, that means it might be more complex than that...

For future readers, an interesting experiment would be to see if the issue persists whenever they update the kernel (but I doubt they will, previous experience showed that they rarely backported fixes from newer devices).
